From c2da0d3811222337d012aab078a503ff683e72e3 Mon Sep 17 00:00:00 2001
From: Bin Han <bin.b.han@intel.com>
Date: Mon, 29 Feb 2016 09:53:09 +0800
Subject: [PATCH 4/5] ov8858_btns: Add new 2760x1560 video setting

This setting is to replace the 1080P setting for 1080P
video recording use case,  because it has larger FOV and
ISP will does 1.43X YUV downscaling to 1080P

Original-Tracked_On: #H1504169207
Change-Id: Ic18281e7b16152eab0bfde775d8ee91ec31bb1fd
Signed-off-by: Bin Han <bin.b.han@intel.com>
Reviewed-on: https://icggerrit.ir.intel.com/104682
Approver: Jing Tao <jing.tao@intel.com>
Reviewed-by: HaiqiangX Yao <haiqiangx.yao@intel.com>
Reviewed-by: DaleiX Liu <daleix.liu@intel.com>
Validation-Android: Wei Tao <wei.tao@intel.com>
Build: Dessert M BXT RVP (SC)
Integrator: WeiX Meng <weix.meng@intel.com>
Build: Dessert M CHT FFD (SC)
Reviewed-by: WeiX Meng <weix.meng@intel.com>
Tracked-On: https://jira01.devtools.intel.com/browse/OAM-22830
Signed-off-by: Tao Wei <wei.tao@intel.com>
Reviewed-on: https://android.intel.com:443/480581
---
 .../camera/drivers/media/i2c/ov8858_btns.h         | 64 ++++++++++++++++++++--
 1 file changed, 59 insertions(+), 5 deletions(-)

diff --git a/drivers/external_drivers/camera/drivers/media/i2c/ov8858_btns.h b/drivers/external_drivers/camera/drivers/media/i2c/ov8858_btns.h
index c37dd2c..ab9cc66 100644
--- a/drivers/external_drivers/camera/drivers/media/i2c/ov8858_btns.h
+++ b/drivers/external_drivers/camera/drivers/media/i2c/ov8858_btns.h
@@ -902,6 +902,60 @@ static const struct ov8858_reg ov8858_2704x2032[] = {
 	{OV8858_TOK_TERM, 0, 0}
 };
 
+static const struct ov8858_reg ov8858_2760x1560[] = {
+	{OV8858_8BIT, 0x0100, 0x00}, /* software_standby */
+	{OV8858_8BIT, 0x3768, 0x22},
+	{OV8858_8BIT, 0x3769, 0x66},
+	{OV8858_8BIT, 0x376a, 0x24},
+	{OV8858_8BIT, 0x3841, 0xFF},
+	{OV8858_8BIT, 0x3800, 0x00}, /* h_crop_start high */
+	{OV8858_8BIT, 0x3801, 0x0C}, /* h_crop_start low */
+	{OV8858_8BIT, 0x3802, 0x00}, /* v_crop_start high */
+	{OV8858_8BIT, 0x3803, 0x0C}, /* v_crop_start low */
+	{OV8858_8BIT, 0x3804, 0x0C}, /* h_crop_end high */
+	{OV8858_8BIT, 0x3805, 0xD3}, /* h_crop_end low */
+	{OV8858_8BIT, 0x3806, 0x09}, /* v_crop_end high */
+	{OV8858_8BIT, 0x3807, 0xA3}, /* v_crop_end low */
+	{OV8858_8BIT, 0x3808, 0x0A}, /* h_output_size high 2704 x 2032 */
+	{OV8858_8BIT, 0x3809, 0xC8}, /* h_output_size low */
+	{OV8858_8BIT, 0x380A, 0x06}, /* v_output_size high */
+	{OV8858_8BIT, 0x380B, 0x18}, /* v_output_size low */
+	{OV8858_8BIT, 0x380C, 0x07}, /* horizontal timing size high */
+	{OV8858_8BIT, 0x380D, 0x90}, /* horizontal timing size low */
+	{OV8858_8BIT, 0x380E, 0x09}, /* vertical timing size high */
+	{OV8858_8BIT, 0x380F, 0xAE}, /* vertical timing size low */
+	{OV8858_8BIT, 0x3810, 0x00},
+	{OV8858_8BIT, 0x3811, 0x04},
+	{OV8858_8BIT, 0x3812, 0x00},
+	{OV8858_8BIT, 0x3813, 0x02},
+	{OV8858_8BIT, 0x3814, 0x01}, /* h_odd_inc */
+	{OV8858_8BIT, 0x3815, 0x01}, /* h_even_inc */
+	{OV8858_8BIT, 0x3821, 0x46}, /* format2 */
+	{OV8858_8BIT, 0x382A, 0x01}, /* v_odd_inc */
+	{OV8858_8BIT, 0x382B, 0x01}, /* v_even_inc */
+	{OV8858_8BIT, 0x382D, 0xFF},
+
+	{OV8858_8BIT, 0x4001, 0x00},
+	{OV8858_8BIT, 0x4022, 0x0A},
+	{OV8858_8BIT, 0x4023, 0x5A},
+	{OV8858_8BIT, 0x4024, 0x0F},
+	{OV8858_8BIT, 0x4025, 0x36},
+	{OV8858_8BIT, 0x4026, 0x0F},
+	{OV8858_8BIT, 0x4027, 0x37},
+	{OV8858_8BIT, 0x4028, 0x00},
+	{OV8858_8BIT, 0x4029, 0x04},
+	{OV8858_8BIT, 0x402a, 0x04},
+	{OV8858_8BIT, 0x402b, 0x08},
+	{OV8858_8BIT, 0x402c, 0x00},
+	{OV8858_8BIT, 0x402d, 0x02},
+	{OV8858_8BIT, 0x402e, 0x04},
+	{OV8858_8BIT, 0x402f, 0x08},
+	{OV8858_8BIT, 0x4600, 0x01}, /* Unknown */
+	{OV8858_8BIT, 0x4601, 0x57}, /* Unknown */
+	{OV8858_8BIT, 0x5901, 0x00},
+
+	{OV8858_TOK_TERM, 0, 0}
+};
 
 static const struct ov8858_reg ov8858_1080P[] = {
 	{OV8858_8BIT, 0x0100, 0x00}, /* software_standby */
@@ -1380,12 +1434,12 @@ static struct ov8858_resolution ov8858_res_video[] = {
 			}
 		},
 	},
-    {
-		.desc = "ov8858_1080P_VIDEO",
-		.width = 1936,
-		.height = 1096,
+	{
+		.desc = "ov8858_2760x1560_VIDEO",
+		.width = 2760,
+		.height = 1560,
 		.used = 0,
-		.regs = ov8858_1080P,
+		.regs = ov8858_2760x1560,
 		.bin_factor_x = 0,
 		.bin_factor_y = 0,
 		.skip_frames = 1,
-- 
1.9.1

